test/wrappedNodeFetch.test.ts

```diff
@@ -0,0 +1,108 @@
+import { wrappedNodeFetch } from '../integrations/octokit/require';
+import { Response } from 'node-fetch';
+import { createExecutionContext } from '../src/context';
+describe('wrappedNodeFetch', () => {
+  it('should call fetch function with correct arguments in record mode', async () => {
+    const mockFetch = jest.fn().mockResolvedValueOnce(new Response());
+    const ctx = {
+      context:{
+      mode: 'record',
+      testId: 'testId',
+      mocks: [],
+      deps: [],
+      }
+    };
+    createExecutionContext(ctx)
+    const wrappedFetch = (wrappedNodeFetch(mockFetch) as any).bind({ fetch: mockFetch });
+    const url = 'http://example.com';
+    const options = {
+      method: 'GET',
+    };
+    const response = await wrappedFetch(url, options);
+    expect(mockFetch).toHaveBeenCalledWith(url, options);
+    expect(response).toBeInstanceOf(Response);
```

Comment:
Add a expect check for updated context after fetch call. to check that whether output of fetch call is added to deps/mocks or not.

Reply:
Added expect check @re-Tick

Reply:
Please add a check to compare the recorded output with response. This will ensure the functionality of wrappedNodeFetch.

Reply:
Added @re-Tick

Reply:
Awesome. LGTM. Resolving the requested changes